src/main/kotlin/io/kuzzle/sdk/controllers/CollectionController.kt

```diff
@@ -0,0 +1,203 @@
+      .query(KuzzleMap().apply {
+        put("controller", "collection")
+        put("action", "create")
+        put("index", index)
+        put("collection", collection)
+        put("mapping", mapping)
+      })
+      .thenApplyAsync { response -> (response.result as ConcurrentHashMap<String, Any?>)["acknowledged"] as Boolean}
+  }
+
+  fun delete(
+      index: String,
+      collection: String
+    ): CompletableFuture<Void> {
+    return kuzzle
+      .query(KuzzleMap().apply {
+        put("controller", "collection")
+        put("action", "delete")
+        put("index", index)
+        put("collection", collection)
+      })
+      .thenApplyAsync { null }
+  }
+
+  fun deleteSpecifications(
+      index: String,
+      collection: String
+    ): CompletableFuture<Void> {
+    return kuzzle
+      .query(KuzzleMap().apply {
+        put("controller", "collection")
+        put("action", "deleteSpecifications")
+        put("index", index)
+        put("collection", collection)
+      })
+      .thenApplyAsync { null }
+  }
+
+  fun exists(
+      index: String,
+      collection: String
+    ): CompletableFuture<Boolean> {
+    return kuzzle
+      .query(KuzzleMap().apply {
+        put("controller", "collection")
+        put("action", "exists")
+        put("index", index)
+        put("collection", collection)
+      })
+      .thenApplyAsync { response -> response.result as Boolean }
+  }
+
+  fun getMapping(
+    index: String,
+    collection: String
+  ): CompletableFuture<ConcurrentHashMap<String, Any?>> {
+  return kuzzle
+    .query(KuzzleMap().apply {
+      put("controller", "collection")
+      put("action", "getMapping")
+      put("index", index)
+      put("collection", collection)
+    })
+    .thenApplyAsync { response -> response.result as ConcurrentHashMap<String, Any?> }
+}
+
+fun getSpecifications(
+    index: String,
+    collection: String
+  ): CompletableFuture<ConcurrentHashMap<String, Any?>> {
+  return kuzzle
+    .query(KuzzleMap().apply {
+      put("controller", "collection")
+      put("action", "getSpecifications")
+      put("index", index)
+      put("collection", collection)
+    })
+    .thenApplyAsync { response -> response.result as ConcurrentHashMap<String, Any?> }
+}
+
+  fun list(index: String): CompletableFuture<ConcurrentHashMap<String, Any?>> {
+    return kuzzle
+        .query(KuzzleMap().apply {
+          put("controller", "collection")
+          put("action", "list")
+          put("index", index)
+        })
+        .thenApplyAsync { response -> response.result as ConcurrentHashMap<String, Any?> }
+  }
+
+  fun refresh(
+    index: String,
+    collection: String
+   ): CompletableFuture<Void> {
+    return kuzzle
+        .query(KuzzleMap().apply {
+          put("controller", "collection")
+          put("action", "refresh")
+          put("index", index)
+          put("collection", collection)
+        })
+        .thenApplyAsync { null }
+  }
+
+  @JvmOverloads
```

Comment:
What does this annotation mean?

Reply:
Since they are several optional parameters, this annotation is for the compiler to generate overloads for this functions (for Java)